src/Model/ModeManager.re

```diff
@@ -0,0 +1,15 @@
+let current: State.t => Oni_Core.Mode.t =
+  (state: State.t) =>
+    if (Selectors.terminalIsActive(state)) {
+      TerminalInsert;
+    } else {
+      switch (state.mode) {
```

Comment:
I think it would be a good idea to rename `state.mode` to `state.vimMode`. Otherwise this is going to cause a lot of confusion I think.

Reply:
Good idea 👍